### PR TITLE
Fix function active/archived tab switching

### PR DESCRIPTION
**Pull request by goodoldneon**
## Description
Fix stale data when switching between active and archived functions

## Type of change (choose one)
- [ ] Chore (refactors, upgrades, etc.)
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ ] I've linked any associated issues to this PR.
- [ ] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/useRows.ts [hunk 1]
```diff
@@ -44,7 +44,12 @@ export function useRows({ archived }: { archived: boolean }): Output {
 
   // Append new function data.
   useEffect(() => {
-    if (functionsRes.data && functionsRes.data.page.page > functionsData.lastPage) {
+    if (!functionsRes.data || functionsRes.isLoading) {
+      return;
+    }
+
+    if (functionsRes.data.page.page > functionsData.lastPage) {
+      console.log('a2', functionsRes.data.functions.length);
```
Comment (tonyhb) [inline ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/useRows.ts:52]:
```suggestion
```

**Review by tonyhb — DISMISSED** [review]

File: ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/useRows.ts (hunk 1 above)
Comment (goodoldneon) [reply ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/useRows.ts:52]: Nooooo I thought I found all those. Thanks, Mr. Eagle Eye

File: ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/useRows.ts [hunk 2]
```diff
@@ -44,7 +44,11 @@ export function useRows({ archived }: { archived: boolean }): Output {
 
   // Append new function data.
   useEffect(() => {
-    if (functionsRes.data && functionsRes.data.page.page > functionsData.lastPage) {
+    if (!functionsRes.data || functionsRes.isLoading) {
+      return;
+    }
+
+    if (functionsRes.data.page.page > functionsData.lastPage) {
```
Comment (djfarrelly) [inline ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/useRows.ts:51]: How does this know to `setFunctionsData` on only archived vs. not archived?

**Review by djfarrelly — APPROVED** [review]
Left a single question, but I can see how the existing change looks like a valid and correct fix!

File: ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/useRows.ts (hunk 2 above)
Comment (goodoldneon) [reply ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/[environmentSlug]/functions/useRows.ts:51]: The data is cleared out when `archived` changes:
https://github.com/inngest/inngest/blob/main/ui/apps/dashboard/src/app/(organization-active)/(dashboard)/env/%5BenvironmentSlug%5D/functions/useRows.ts#L33

This whole hook needs to be refactored. It's probably the worst code I've written here

**Pull request merged**